ui/app/src/components/PreviewComponentsPanel/PreviewComponentsPanel.tsx

```diff
@@ -55,51 +57,63 @@ const translations = defineMessages({
   listInPageInstances: {
     id: 'previewComponentsPanel.listInPageInstances',
     defaultMessage: 'Instances on this page'
+  },
+  filter: {
+    id: 'previewComponentsPanel.filter',
+    defaultMessage: 'Filter...'
   }
 });
 
-type ComponentsPanelUIProps = PropsWithResource<ContentType[]>;
+type ComponentsPanelUIProps = { componentTypes: ContentType[] };
 
 export function PreviewComponentsPanel() {
-  const resource = useSelectorResource<ContentType[], EntityState<ContentType>>((state) => state.contentTypes, {
-    shouldRenew: (source, resource) => resource.complete,
-    shouldResolve: (source) => !source.isFetching && nnou(source.byId),
-    shouldReject: (source) => nnou(source.error),
-    errorSelector: (source) => source.error,
-    resultSelector: (source) =>
-      Object.values(reversePluckProps(source.byId, '/component/level-descriptor')).filter(
-        (contentType) => contentType.type === 'component'
+  const contentTypesBranch = useSelection((state) => state.contentTypes);
```

Comment:
We're doing `Object.values|keys` a couple of times throughout this function body. Might as well cache and use everywhere needed.
```suggestion
  const contentTypesBranch = useSelection((state) => state.contentTypes);
  const contentTypesBranchEntries = Object.entries(contentTypesBranch.byId ?? {});
```